### ISSUE TITLE
Clarification on the return of Listing available DGGRS (/dggs)  (section 6.2)

### ISSUE THREAD

**Issue by tik65536** [opengeospatial/ogcapi-discrete-global-grid-systems#79]
6.2 (/dggs) [Requirement 1D](https://docs.ogc.org/DRAFTS/21-038.html#_listing_available_dggrs_dggs)
- is uri duplicated with links, or can it be included in the links and remove the need for URI?
 
6.2 (/dggs) [Requirement 1E](https://docs.ogc.org/DRAFTS/21-038.html#_listing_available_dggrs_dggs)
- The links property here should more clearly refer to the elements inside the dggrs list. It confuses with the links of dggrs-list

The links property of dggrs-list return (not under the elements of dggrs list) is not quite defined here. Is it the last one (Requirement 1F) that we should include [ogc-rel:dggrs-list]? But from the sample [C.2 ](https://docs.ogc.org/DRAFTS/21-038.html#_listing_available_dggrss), it doesn't show that the dggrs-list is included, maybe because it is from collections?

Thanks 
Tik

**Comment by jerstlouis** [comment]
Hi @tik65536 ,

> is uri duplicated with links, or can it be included in the links and remove the need for URI?

Short answer: no.

The link with relation type `[ogc-rel:dggrs-definition]` serves a different purpose than the URI.

The URI is for a DGGRS registered in an authorittive register, such as the DGGS register that we hope to set up on the OGC Definition Server (#36). The URI is only mandatory if the DGGS has already been officially registered somewhere. Ideally, a JSON representation of the resource at that link should return the DGGRS definition as per Annex B, but this is not guaranteed. The URI is compared against known DGGRS by a simple string comparison.

The `[ogc-rel:dggrs-definition]` link is always mandatory, and would always return the JSON definition as per the Annex B JSON schema. Ideally, the link to definition would be to a definition hosted by the local implementation, though if the URI on the DGGRS register does return a proper DGGRS JSON definition, then it could point to that. The advantage of using a local resource is that it avoids relying on the availability of the external register.

> The links property here should more clearly refer to the elements inside the dggrs list. It confuses with the links of dggrs-list

I'm not sure I fully understand what you're saying and asking here.

Requirement 1 is all about the response for `.../dggs` (the DGGRS list) (except for 1F).

> a link to _the discrete global grid reference system_ 

This refers to the resources at `.../dggs/{dggrsId}`.
We could probably add _description_ here because this is what this resource is called: the DGGRS _descriptions_.

> It confuses with the links of dggrs-list

Requirement 1E is talking about the `links` inside `.../dggs` (the DGGRS list).
We could clarify that which `links` we're talking about here.

Requirement 1F is talking about links _to_ `.../dggs` (e.g., from the API landing page, or from a specific collection).
We could provide these examples as clarifications here.

**Comment by jerstlouis** [comment]
@tik65536 I also realized there was a mistake in our GNOSIS demo server implementation which I've fixed and will update ASAP. The DGGRS array property in the response for `/dggs` should be called `dggrs`, not `dggs`. [Example C.2](https://docs.ogc.org/DRAFTS/21-038.html#_listing_available_dggrss) was correct.

**Comment by tik65536** [comment]
@jerstlouis , thx for the reply. Thx for the explain and clarification on uri. 

The description of requirements 1E and 1F often confuses me (maybe just me :P ), I would explain more and describe it with the example JSON below. 

For this /dggs API endpoints, we have 2 links properties, one is inside the element of dggrs array (Requirement 1D) , and the other one is at the root level. 

> 1E : The links property SHALL include at minimum a link to the discrete global grid reference system using the self link relation type, as well as a link to the discrete global grid reference system definition using the [ogc-rel:dggrs-definition] relation type.

Is Requirement 1E describing what should be included in the links property of Requirement 1D? The description matched with the example below. 

But what should be included in the root level links property of /dggs , is it related to 1F? 

```json 
{
   # What should be included in the links here? Is it related to 1F? 
   "links" : [
      { "rel" : "[ogc-rel:geodata]", "href" : "/ogcapi/collections/gebco" }
   ],
   "dggrs" : [
      {
         "id" : "GNOSISGlobalGrid",
         "title" : "GNOSIS Global Grid",
         "uri" : "[ogc-dggrs:GNOSISGlobalGrid]",
         #  Requirement 1E ?
         "links" : [  
            { "rel" : "self", "title" : "GNOSISGlobalGrid DGGRS for GEBCO",
              "href" : "/ogcapi/collections/gebco/dggs/GNOSISGlobalGrid" },
            { "rel" : "[ogc-rel:dggrs-definition]", "title" : "GNOSISGlobalGrid DGGRS definition",
              "href" : "/ogcapi/dggrs/GNOSISGlobalGrid"
            }
         ]
      }
```

**Comment by jerstlouis** [comment]
@tik65536 

> Is Requirement 1E describing what should be included in the links property of Requirement 1D? 

Yes. The requirements do not mention the top-level `links` property at all, and in the `/dggs` response the links at the top-level is [not required](https://github.com/opengeospatial/ogcapi-discrete-global-grid-systems/blob/master/openapi/responses/dggs-core/rDGGRSList.yaml#L6) either. We should clarify that.

> What should be included in the links here? Is it related to 1F? 

Whichever link you want to include. In the example it is a link back to the origin resource to which this `.../dggs` end-point is associated (the collection). This `links` property is optional. Not related to 1F.

> Requirement 1E ?

Requirements D and E are both talking about this `links` property inside each DGGRS element.

Requirement 1F is talking about the link you will find that points **TO** this `.../dggs` end-point.

For example you will find in https://maps.gnosis.earth/ogcapi/collections/gebco?f=json :

```json
      {
         "rel" : "http://www.opengis.net/def/rel/ogc/1.0/dggrs-list",
         "title" : "Discrete Global Grid Systems for General Bathymetric Chart of the Oceans",
         "href" : "/ogcapi/collections/gebco/dggs"
      },
```

[Requirement 1F](https://docs.ogc.org/DRAFTS/21-038.html#_req_core_dggrs-list)

> The link relation type to use for linking from the corresponding resource **to the list of available DGGRSs** for that resource SHALL be `[ogc-rel:dggrs-list]`

**Comment by tik65536** [comment]
Ok, Thx. 
Let's clarify this at Wednesday's meeting.

**Comment by jerstlouis** [comment]
SWG 2025-01-08:

We need to make the following changes to Requirement 1 to complete this issue:
- We could probably add _description_ in Requirement 1 because this is what this resource is called: the DGGRS descriptions.
- Requirement 1D: The requirements do not mention the top-level links property at all, and in the /dggs response the links at the top-level is [not required](https://github.com/opengeospatial/ogcapi-discrete-global-grid-systems/blob/master/openapi/responses/dggs-core/rDGGRSList.yaml#L6) either. We should clarify that.- Requirement 1E: We could clarify which links we're talking about here.
- Requirement 1F: is talking about links to .../dggs (e.g., from the API landing page, or from a specific collection).
We could provide these examples as clarifications here.
